common/src/main/kotlin/io/homeassistant/companion/android/common/data/network/NetworkStatusMonitor.kt

```diff
@@ -0,0 +1,71 @@
+package io.homeassistant.companion.android.common.data.network
+
+import android.net.ConnectivityManager
+import android.net.Network
+import android.net.NetworkCapabilities
+import android.net.NetworkRequest
+import io.homeassistant.companion.android.database.server.ServerConnectionInfo
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.flow.distinctUntilChanged
+
+interface NetworkStatusMonitor {
+    /**
+     * Observes network state changes. Emits current state immediately.
+     * @return Flow of NetworkState updates
+     */
+    fun observeNetworkStatus(serverConfig: ServerConnectionInfo): Flow<NetworkState>
+}
+
+/**
+ * Tracks network changes using ConnectivityManager callbacks.
+ * Automatically handles callback registration/cleanup.
+ */
+@Singleton
+class NetworkStatusMonitorImpl @Inject constructor(
```

Comment:
```suggestion
@Singleton
internal class NetworkStatusMonitorImpl @Inject constructor(
```

No need to expose the impl outside of `:common`. Still the same question do we need it to be a Singleton.